backend/migrations/007_add_multi_source_support.py

```diff
@@ -0,0 +1,56 @@
+"""Add multi-source translation support tables."""
+import peewee as pw
+
+# Use proxy pattern like other migrations to avoid model conflicts
+db_proxy = pw.DatabaseProxy()
+
+class _Base(pw.Model):
+    class Meta:
+        database = db_proxy
+
+class SourceTranslationLinks(_Base):
+    id = pw.AutoField(primary_key=True)
+    origin_source_id = pw.IntegerField()
+    translated_source_id = pw.IntegerField()
+
+    class Meta:
+        database = db_proxy
+        table_name = 'source_translation_links'
+        indexes = (
+            (('origin_source_id',), False),
+            (('translated_source_id',), False),
+        )
+
+class SegmentTranslationLinks(_Base):
```

Comment:
We need another field for this table. Some kind of "type" field / enum with options:
1. Origin
2. Translated

We can also do this by marking the origin sources/segment with a property / type saying it is origin. Now sure what is better.